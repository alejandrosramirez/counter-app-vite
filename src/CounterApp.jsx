import { useState } from "react";
import PropTypes from "prop-types";

export const CounterApp = ({ value }) => {
	const [counter, setCounter] = useState(value);
	const handleAdd = () => setCounter(counter + 1);
	const handleSubstract = () => setCounter(counter - 1);
	const handleReset = () => setCounter(value);

	return (
		<>
			<h1>Counter App</h1>
			<h2>{counter}</h2>

			<button aria-label="btn-add" onClick={handleAdd}>+1</button>
			<button aria-label="btn-sub" onClick={handleSubstract}>-1</button>
			<button aria-label="btn-res" onClick={handleReset}>Reset</button>
		</>
	);
};

CounterApp.propTypes = {
	value: PropTypes.number.isRequired,
};
