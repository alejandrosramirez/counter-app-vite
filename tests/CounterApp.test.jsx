import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe("Pruebas en <CounterApp />", () => {
	const value = 10;

	test("Debe de hacer match con el snapshot", () => {
		const { container } = render(<CounterApp value={value} />);
		expect(container).toMatchSnapshot();
	});

	test("Debe de mostrar el valor inicial de 100", () => {
		render(<CounterApp value={100} />);
		expect(screen.getByText(100)).toBeTruthy();
		expect(screen.getByRole("heading", { level: 2 }).innerHTML).toContain("100");
	});

	test("Debe de incrementar con el botón +1", () => {
		render(<CounterApp value={value} />);
		fireEvent.click(screen.getByRole("button", { name: "btn-add" }));
		expect(screen.getByText("11")).toBeTruthy();
	});

	test("Debe de decrementar con el botón -1", () => {
		render(<CounterApp value={value} />);
		fireEvent.click(screen.getByRole("button", { name: "btn-sub" }));
		expect(screen.getByText("9")).toBeTruthy();
	});

	test("Debe de reiniciar al valor por defecto con el botón reset", () => {
		render(<CounterApp value={value} />);
		fireEvent.click(screen.getByRole("button", { name: "btn-add" }));
		fireEvent.click(screen.getByRole("button", { name: "btn-add" }));
		fireEvent.click(screen.getByRole("button", { name: "btn-add" }));
		fireEvent.click(screen.getByRole("button", { name: "btn-res" }));
		expect(screen.getByText("10")).toBeTruthy();
	});
});
