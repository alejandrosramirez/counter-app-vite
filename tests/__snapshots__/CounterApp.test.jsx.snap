// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    10
  </h2>
  <button
    aria-label="btn-add"
  >
    +1
  </button>
  <button
    aria-label="btn-sub"
  >
    -1
  </button>
  <button
    aria-label="btn-res"
  >
    Reset
  </button>
</div>
`;
